components/intercom/conversation-closed.js

```diff
@@ -0,0 +1,54 @@
+const intercom = require("https://github.com/PipedreamHQ/pipedream/components/intercom/intercom.app.js");
+
+module.exports = {
+  name: "New Closed Conversation",
+  description: "Emits an event each time a conversation is closed.",
+  version: "0.0.1",
+  dedupe: "unique",
+  props: {
+    intercom,
+    db: "$.service.db",
+    timer: {
+      type: "$.interface.timer",
+      default: {
+        intervalSeconds: 60 * 15,
+      },
+    },
+  },
+
+  async run(event) {
+    const monthAgo = this.intercom.monthAgo();
+    let lastConversationClosedAt =
+      this.db.get("lastConversationClosedAt") || Math.floor(monthAgo / 1000);
+    const data = {
```

Comment:
In this and each of the components below, I'd recommend including the `data` object, as well as the code for filtering on objects since the last item (e.g. `lastConversationClosedAt`), within the app file method itself - here, `this.intercom.searchConversations`.

That is, it'd be great if the component itself just contained this code in the run method:

```javascript
const monthAgo = this.intercom.monthAgo();
let lastConversationClosedAt = this.db.get("lastConversationClosedAt") || Math.floor(monthAgo / 1000);
const conversations = await this.intercom.searchConversations(data, starting_after, lastConversationClosedAt);
for (const conversation of conversations) {
  this.$emit(conversation, {
    id: `${conversation.id}${conversation.last_close_at}`,
    summary: conversation.source.body,
    ts: conversation.last_close_at,
  });
}
```

By passing `lastConversationClosedAt` to the app file method, I think you can handle all of the other logic in the app file, and just return the conversations that meet your criteria.

I think you can do the same in each of the components below - just a little refactor.